clients/src/main/java/org/apache/kafka/common/protocol/SendBuilder.java

```diff
@@ -137,6 +138,9 @@ public void writeRecords(BaseRecords records) {
         if (records instanceof MemoryRecords) {
             flushPendingBuffer();
             addBuffer(((MemoryRecords) records).buffer());
+        } else if (records instanceof UnalignedMemoryRecords) {
```

Comment:
Are there any tests we can update to cover this?

Reply:
Add a testZeroCopyUnalignedRecords